src/test/scala/zio/interop/TwitterSpec.scala

```diff
@@ -1,82 +1,71 @@
 package zio.interop
 
+import com.twitter.util.{ Await, Future, FuturePool }
 import java.util.concurrent.atomic.AtomicInteger
-
-import com.twitter.util.{ Await, Future, Promise }
-import zio.Task
+import zio._
+import zio.duration._
 import zio.interop.twitter._
 import zio.test._
 import zio.test.Assertion._
-import zio.test.TestAspect.flaky
-
-import scala.util.{ Failure, Success, Try }
+import zio.test.TestAspect.nonFlaky
+import zio.test.environment.Live
 
 object TwitterSpec extends DefaultRunnableSpec {
   val runtime = runner.runtime
 
-  override def spec =
+  def spec =
     suite("TwitterSpec")(
       suite("Task.fromTwitterFuture")(
-        testM("return failing `Task` if future failed.") {
-          val error  = new Exception
-          def future = Future.exception[Int](error)
-          val task   = Task.fromTwitterFuture(future).unit
-
-          assertM(task.either)(isLeft(equalTo(error)))
+        testM("lifts failed futures") {
+          for {
+            error  <- UIO(new Exception)
+            result <- Task.fromTwitterFuture(Future.exception(error)).either
+          } yield assert(result)(isLeft(equalTo(error)))
         },
-        testM("return successful `Task` if future succeeded.") {
-          val value  = 10
-          def future = Future.value(value)
-          val task   = Task.fromTwitterFuture(future).option
-
-          assertM(task)(isSome(equalTo(value)))
+        testM("lifts successful futures") {
+          for {
+            value  <- UIO(10)
+            result <- Task.fromTwitterFuture(Future.value(value))
+          } yield assert(result)(equalTo(value))
         },
-        testM("ensure future is interrupted together with task.") {
-          val value = new AtomicInteger(0)
-
-          val promise = new Promise[Unit] with Promise.InterruptHandler {
-            override protected def onInterrupt(t: Throwable): Unit = setException(t)
-          }
-
-          def future = promise.flatMap(_ => Future(value.incrementAndGet()))
+        testM("ensures future is interrupted") {
+          def infiniteFuture(ref: AtomicInteger): Future[Nothing] =
+            FuturePool.interruptibleUnboundedPool(ref.getAndIncrement()).flatMap(_ => infiniteFuture(ref))
 
           for {
-            fiber <- Task.fromTwitterFuture(future).fork
+            ref   <- UIO(new AtomicInteger(0))
+            fiber <- Task.fromTwitterFuture(infiniteFuture(ref)).fork
             _     <- fiber.interrupt
-            _     <- Task.effect(promise.setDone())
-            a     <- fiber.await
-            v     <- Task.effectTotal(value.get)
-          } yield assert(a.toEither)(isLeft) && assert(v)(isZero)
-        }
+            _     <- Live.live(clock.sleep(20.millis))
```

Comment:
Without this sleep, the test becomes flaky. I presume it's caused by interrupt signal propagation, so please double-check my assumption and suggest an alternative if this is not an acceptable approach.